pennylane/drawer/utils.py

```diff
@@ -57,3 +58,44 @@ def convert_wire_order(tape, wire_order=None, show_all_wires=False):
         wire_order = [wire for wire in wire_order if wire in used_wires]
 
     return {wire: ind for ind, wire in enumerate(wire_order)}
+
+
+def unwrap_controls(op):
+    """Unwraps nested controlled operations for drawing.
+
+    Controlled operations may themselves contain controlled operations; check
+    for any nesting of operators when drawing so that we correctly identify
+    and label _all_ control and target qubits.
+
+    Args:
+        op (.Operation): A PennyLane operation.
+
+    Returns:
+        Wires, List: The control wires of the operation, along with any associated
+        control values.
+    """
+    control_wires = getattr(op, "control_wires", [])
+
+    # Get wires and control values of base operation; need to make a copy of
+    # control values, otherwise it will modify the list in the operation itself.
+    control_values = op.hyperparameters.get("control_values", None)
+    if isinstance(control_values, list):
+        control_values = control_values.copy()
```

Comment:
```suggestion
    # Get wires and control values of base operation; need to make a copy of
    # control values, otherwise it will modify the list in the operation itself.
    control_wires = getattr(op, "control_wires", [])
    control_values = op.hyperparameters.get("control_values", None)

    if isinstance(control_values, list):
        control_values = control_values.copy()
```